nifi-nar-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/main/java/org/apache/nifi/snmp/processors/SendTrapSNMP.java

```diff
@@ -147,17 +158,18 @@ public void onTrigger(final ProcessContext context, final ProcessSession process
                 attributes.put("enterpriseOid", enterpriseOid);
                 attributes.put("genericTrapType", genericTrapType);
                 attributes.put("specificTrapType", specificTrapType);
-                snmpHandler.sendTrap(attributes, v1TrapConfiguration);
+                snmpHandler.sendTrap(attributes, v1TrapConfiguration, target);
             } else {
                 final String trapOidValue = context.getProperty(V2TrapProperties.TRAP_OID_VALUE).evaluateAttributeExpressions(flowFile).getValue();
                 V2TrapConfiguration v2TrapConfiguration = new V2TrapConfiguration(trapOidValue);
                 attributes.put("trapOidValue", trapOidValue);
-                snmpHandler.sendTrap(attributes, v2TrapConfiguration);
+                snmpHandler.sendTrap(attributes, v2TrapConfiguration, target);
             }
-
-            processSession.putAllAttributes(flowFile, attributes);
+            flowFile = processSession.putAllAttributes(flowFile, attributes);
             processSession.transfer(flowFile, REL_SUCCESS);
-
+            if (isFlowFileReceived) {
+                processSession.getProvenanceReporter().receive(flowFile, "/sendTrap");
+            }
```

Comment:
I think we don't need to add a provenance event explicitly. No data (not even attribute) is getting value from any outside sources. Instead, an ATTRIBUTES_UPDATED will be generated automatically.

```suggestion
```

Reply:
The data is not from an external system, I agree, removing.